code/integration-tests/runtime-tests/test/tests/pablo/constantProductDex/constantProductTests.ts

```diff
@@ -1,285 +1,1888 @@
+      // Arbitrary wait because the last test will fail immediately, and can cause this test to run
+      // into priority too low issues.
+      await waitForBlocks(api);
+      const exc = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.swap(
+          hardCodedPool1.poolId,
+          { assetId: 4, amount: Pica(100) },
+          {
+            assetId: 4,
+            amount: Pica(100)
+          },
+          false
+        )
+      ).catch(exc => exc);
+      expect(exc.toString()).to.contain("pablo.CannotRespectMinimumRequested");
+    });
+
+    it("#4.11 I can not buy or swap in a pool that doesn't exist.", async function () {
+      // Arbitrary wait because the last test will fail immediately, and can cause this test to run
+      // into priority too low issues.
+      await waitForBlocks(api);
+      const err = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.buy(1337, 131, { assetId: 1, amount: 10_000 }, false)
+      ).catch(exc => exc);
+      expect(err.toString()).to.contain("pablo.PoolNotFound");
+      const err2 = await sendAndWaitForSuccess(
+        api,
+        walletTrader1,
+        api.events.pablo.Swapped.is,
+        api.tx.pablo.swap(1337, { assetId: 131, amount: 1_000 }, { assetId: 1, amount: 10_000 }, false)
+      ).catch(exc => exc);
+      expect(err2.toString()).to.contain("pablo.PoolNotFound");
+    });
 
     it(
-      "Given that the pool has liquidity and the users have LPTokens, " +
-        "users can remove liquidity from the pool by using LP Tokens",
+      "#4.12 I can buy or swap with the minimum amount requested greater than the trade would give, " +
+        "but the amounts will be automatically adjusted to the pools ratio.",
       async function () {
-        const result = await removeLiquidityFromPool(api, poolId, walletId1, Pica(500));
-        // ToDo (D. Roth): Re- enable after pablo rework!
-        // expect(result.resultBase.toBn()).to.be.bignumber.closeTo(result.resultQuote.toBn(), Pica(15).toString());
+        const exc = await sendAndWaitForSuccess(
+          api,
+          walletTrader1,
+          api.events.pablo.Swapped.is,
+          api.tx.pablo.swap(
+            hardCodedPool1.poolId,
+            { assetId: hardCodedPool1.quoteAssetId, amount: 10 },
+            { assetId: hardCodedPool1.baseAssetId, amount: 2000_000 },
+            false
+          )
+        ).catch(exc => exc);
+        expect(exc.toString()).to.contain("pablo.CannotRespectMinimumRequested");
       }
     );
 
-    it("Given that LPTokens are transferred to another user, other user can removeLiquidity", async function () {
-      const result = await removeLiquidityFromPool(
+    it("#4.13 I can buy a huge amount with very high slippage.", async function () {
```

Comment:
Just to have a standard outlook throughout the tests, a simple assertion would be goo here as well.

Reply:
Thanks for seeing this. I forgot to finalize this test. 👀